nobl9/anomaly_config_schema.go

```diff
@@ -0,0 +1,190 @@
+		),
+	)
+}
+
+func marshalAnomalyConfig(anomalyConfigRaw interface{}) *n9api.AnomalyConfig {
+	anomalyConfigSet := anomalyConfigRaw.(*schema.Set)
+	if anomalyConfigSet.Len() == 0 {
+		return nil
+	}
+
+	anomalyConfig := anomalyConfigSet.List()[0].(map[string]interface{})
+	noDataAnomalyConfig := anomalyConfig["no_data"].(*schema.Set).List()[0].(map[string]interface{})
+	noDataAlertMethods := noDataAnomalyConfig["alert_method"].([]interface{})
+
+	return &n9api.AnomalyConfig{
+		NoData: &n9api.AnomalyConfigNoData{
+			AlertMethods: marshalAnomalyConfigAlertMethods(noDataAlertMethods),
+		},
+	}
+}
+
+func marshalAnomalyConfigAlertMethods(alertMethodsTF []interface{}) []n9api.AnomalyConfigAlertMethod {
+	alertMethodsAPI := make([]n9api.AnomalyConfigAlertMethod, len(alertMethodsTF))
+	for i := 0; i < len(alertMethodsTF); i++ {
+		alertMethodTF := alertMethodsTF[i].(map[string]interface{})
+		alertMethodsAPI[i] = n9api.AnomalyConfigAlertMethod{
+			Name:    alertMethodTF["name"].(string),
+			Project: alertMethodTF["project"].(string),
+		}
+	}
+
+	return alertMethodsAPI
+}
+
+func unmarshalAnomalyConfig(d *schema.ResourceData, spec map[string]interface{}) error {
+	anomalyConfigRaw, _ := spec["anomalyConfig"]
+	anomalyConfig := anomalyConfigRaw.(map[string]interface{})
+
+	noData := anomalyConfig["noData"].(map[string]interface{})
+	noDataMethods := noData["alertMethods"].([]interface{})
+	resNoDataMethods := make([]map[string]interface{}, len(noDataMethods))
+
+	for i, amRaw := range noDataMethods {
+		am := amRaw.(map[string]interface{})
+
+		resNoDataMethods[i] = map[string]interface{}{
+			"name":    am["name"],
+			"project": am["project"],
+		}
+	}
+
+	anomalyConfigTF := map[string]interface{}{
+		"no_data": schema.NewSet(oneElementSet, []interface{}{
+			map[string]interface{}{
+				"alert_method": resNoDataMethods,
+			},
+		}),
+	}
+
+	return d.Set(
+		"anomaly_config",
+		schema.NewSet(oneElementSet, []interface{}{anomalyConfigTF}),
+	)
+}
+
+// transformAnomalyConfigAlertMethodsTo2DMap transforms a slice of
+// AnomalyConfigAlertMethod values into a 2D map, where each row of
+// the map corresponds to an alert method and contains the method's
+// name and project as key-value pairs.
+// Example input:
+// [
+//
+//	{ Name: "method1", Project: "project1" },
+//	{ Name: "method2", Project: "project2" },
+//	{ Name: "method3", Project: "project1" },
+//
+// ]
+//
+// Example output:
+//
+//	{
+//	  "method1": { "name": "method1", "project": "project1" },
+//	  "method2": { "name": "method2", "project": "project2" },
+//	  "method3": { "name": "method3", "project": "project1" },
+//	}
+func transformAnomalyConfigAlertMethodsTo2DMap(alertMethods []n9api.AnomalyConfigAlertMethod) map[string]map[string]string {
+	result := make(map[string]map[string]string)
+	for _, method := range alertMethods {
+		values := make(map[string]string)
+
+		values["name"] = method.Name
+		values["project"] = method.Project
+		result[method.Name] = values
```

Comment:
```suggestion
		result[method.Name] = map[string]string{
			"name": method.Name,
			"project": method.Project,
		}
```
For me it's more readable, but you can leave it the way it is, only maybe remove the empty line.